Editor/AGS.Editor/GUI/GUIController.cs

```diff
@@ -261,6 +261,16 @@ public void AddMenuItems(IEditorComponent plugin, MenuCommands commands)
                     {
                         RegisterMenuCommand(command.ID, plugin);
                     }
+
+                    if(command.SubCommands != null && command.SubCommands.Count > 0)
+                    {
+                        foreach (MenuCommand scommand in command.SubCommands)
+                        {
+                            RegisterMenuCommand(scommand.ID, plugin);

```

Comment:
You need to test if scommand.ID is not null, similar to how it's tested for command.ID above.
Otherwise, having separators in submenu would crash the editor.

Reply:
But, on another hand, I just realized, that your submenu handling does not allow nested submenu levels.
Probably this would require recursive call to a helper function, having IList<MenuCommand> as a parameter

Reply:
I used your code and modified AddMenuItem so it could be recursive.